github-runners/terraform/base/main.tf

```diff
@@ -0,0 +1,440 @@
+#Linux x86_64 runner
+resource "aws_instance" "linux_x86_runner" {
+  count             = length(var.public_subnet_cidrs)
+  ami               = data.aws_ami.ubuntu_x86.image_id
+  instance_type     = element(var.instance_type, 1)
+  subnet_id         = element(aws_subnet.public_subnets.*.id, count.index)
+  availability_zone = element(var.azs, count.index)
+  # Security Group
+  vpc_security_group_ids = ["${aws_security_group.allow_runner.id}"]
+  # the Public SSH key
+  key_name                    = var.aws_key_name
+  associate_public_ip_address = true
+
+  tags = {
+    name       = "gh-linux-x86-runner"
+    role       = "runner"
+    os_name    = "ubuntu"
+    os_version = "22.04"
+    arch       = "x86_64"
+  }
+
+  depends_on = [
+    aws_subnet.public_subnets,
+    aws_nat_gateway.nat_gateway,
+    aws_internet_gateway.gw
+  ]
+
+  lifecycle {
+    # ignore_changes = [
+    #   tags, ami
+    # ]
+    create_before_destroy = true
+
+    //ignore_changes = all
+  }
+
+  user_data = <<-EOF
+  #!/bin/bash
+  export DEBIAN_FRONTEND=noninteractive
+  sudo apt update -y
+  sudo apt upgrade -y
+  sudo apt install git curl jq wget gnupg openssl make build-essential jq net-tools -y
+  sudo curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh -s -- --default-toolchain nightly -y
```

Comment:
Why do we install `jq`, `rustup` and a bunch of other tool here> Wouldn't setup below cover everything runner itself needs?

Reply:
I've removed jq and rust will be installed via the pipeline during the job run, so removed that. I removed make and build-essentials.